packages/server/lib/hooks/hooks.ts

```diff
@@ -185,7 +187,9 @@ export const connectionRefreshFailed = async ({
 
     const slackNotificationService = new SlackService({ orchestratorClient: getOrchestratorClient(), logContextGetter });
 
-    void slackNotificationService.reportFailure(connection, connection.connection_id, 'auth', logCtx.id, environment.id, config.provider);
+    const randomDelay = Math.random() * 1000;
+    await delay(randomDelay);
```

Comment:
Adding some delay makes the problem less likely to happen but it doesn't solve the underlying issue

Reply:
True, updated to use mutex